src/ip/Address.cc

```diff
@@ -695,9 +696,11 @@ Ip::Address::InitAddr(struct addrinfo *&ai)
 void
 Ip::Address::FreeAddr(struct addrinfo *&ai)
 {
-    if (ai == nullptr) return;
+    if (!ai)
+        return;
 
-    if (ai->ai_addr) delete ai->ai_addr;
+    if (ai->ai_addr)
+        delete static_cast<struct sockaddr*>(struct sockaddr_in6*>(ai->ai_addr);
```

Comment:
Please keep the two `delete ... ai_addr` calls in sync. I have not studied whether any of them need changing, but if one of them needs changing, then the other one must be changed as well.